app/src/main/java/com/portfolio/tasky/entry/viewModels/LoginViewModel.kt

```diff
@@ -47,7 +56,11 @@ class LoginViewModel @Inject constructor(
         val areFieldsValidated = mutableEmail.value == true && mutablePassword.value == true
         mutableFieldsValid.value = areFieldsValidated  && isNetworkAvailable
     }
-    suspend fun login(authenticationModel: AuthenticationRequest){
-        mutableLogin.postValue(entryRepository.doLogin(authenticationModel))
+    fun login(authenticationModel: AuthenticationRequest){
+        viewModelScope.launch(Dispatchers.IO + coroutineExceptionHandler) {
+            val authenticatedUser = entryRepository.doLogin(authenticationModel)
+            userPreferences.saveAuthenticatedUser(authenticatedUser as AuthenticationResponse)
+            mutableLogin.postValue(entryRepository.doLogin(authenticationModel))
+        }
```

Comment:
Aren't you logging in twice here? Shouldn't it be `mutableLogin.postValue(authenticatedUser)` instead?

Reply:
Yes!!!
and this consumed all my time today :(